tests/runfiles/linux.run

```diff
@@ -151,7 +151,8 @@ tests = ['zfs_rename_001_pos', 'zfs_rename_002_pos', 'zfs_rename_003_pos',
     'zfs_rename_to_encrypted']
 
 [tests/functional/cli_root/zfs_reservation]
-tests = ['zfs_reservation_001_pos', 'zfs_reservation_002_pos']
+tests = ['zfs_reservation_001_pos', 'zfs_reservation_002_pos',
```

Comment:
No need to add a new test file.
In refreserv_005_pos.ksh remove the tests for "# Verify it is affected by volsize" and if you want to add a bigger than test, do it there.

Reply:
Is `refreserv_005_pos.ksh` even working properly? I don't understand why it tries to `zfs set refreservation=- testpool/testfs/vol`, does "-" have a different meaning on Illumos? On ZoL it will always fail with the following error: `cannot set property for 'testpool/testfs/vol': bad numeric value '-'`

source:
```sh
log_must zfs create -V 10M $vol

# Verify the parent filesystem does not affect volume
log_must zfs set quota=25M $fs
log_must zfs set refreservation=10M $vol
avail=$(get_prop mountpoint $vol)
log_mustnot zfs set refreservation=$avail $vol
```

log:
```
ASSERTION: Volume (ref)reservation is not limited by volsize
SUCCESS: zfs create -V 10M testpool/testfs/vol
SUCCESS: zfs set quota=25M testpool/testfs
SUCCESS: zfs set refreservation=10M testpool/testfs/vol
SUCCESS: zfs set refreservation=- testpool/testfs/vol exited 1
```

Reply:
great question! '-' is the response for "not set" and mountpoint is not appropriate for size or volumes. Methinks this should be more like:
```
avail=$(get_prop available $vol)
```
thus avail is the total space available in the pool, which we'll presume is always > the quota on the file system. In reality we just need to test if the refreservation is > the quota, which we set previously. A lazy way would be:
```
log_must zfs set quota=25M $fs
log_must zfs set refreservation=26M $vol
```

but this test is kinda silly, too and knowing the code, this test should only fail if we don't have the available space (which quota doesn't need to know)

methinks a simple test to set refreservation less than and greater than the initial estimate thusly:
```
initial=$(zfs get -p -H -o value refreservation $vol)
log_must zfs set refreservation=$((initial + 1)) $vol
log_must zfs set refreservation=$((initial - 1)) $vol
```
NB, get_prop returns the pretty value, not easy to do math against